awx/main/tasks/receptor.py

```diff
@@ -61,11 +61,17 @@ def read_receptor_config():
             return yaml.safe_load(f)
 
 
-def work_signing_enabled(config_data):
-    for section in config_data:
-        if 'work-verification' in section:
-            return True
-    return False
+def work_signing_enabled(receptor_ctl, remote_node, work_type):
+    receptor_status = receptor_ctl.simple_command("status")
+
+    for node in receptor_status["Advertisements"]:
+        if node["NodeID"] == remote_node:
+            for work in node["WorkCommands"]:
+                if work["WorkType"] == work_type:
+                    return work["Secure"]
+
+    logger.warning(f"{remote_node} not found in receptor advertisments, defaulting to signed work.")
```

Comment:
I'm doing this because you technically submit work to a node that doesn't exist on the mesh yet. I don't know if this will ever happen in the AWX use case, but I just wanted a non-fatal fallback here.